From a0c9c7ac79279c31ff0a6ff9121400f5c94d4288 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Thu, 15 Nov 2012 16:09:07 +0800
Subject: [PATCH 098/304] ENGR00233771 init.rc: fix the kernel tracer sysfs
 permission issue

The debugfs mount should be ahead of the sysfs
file mode change in the init.trace.rc.
So add the debugfs mount into the board init.rc file.

Change-Id: I8032887a71dcecf2993f94584bc49fa258dd0d44
Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 arm2_6dq/init.rc            | 3 +++
 arm2_6sl/init.rc            | 3 +++
 imx53_smd/init.freescale.rc | 3 +++
 imx5x/init.rc               | 3 ---
 imx6/etc/init.rc            | 3 ---
 imx6/sabresd/init.rc        | 3 +++
 6 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/arm2_6dq/init.rc b/arm2_6dq/init.rc
index d51478c..3b92032 100755
--- a/arm2_6dq/init.rc
+++ b/arm2_6dq/init.rc
@@ -55,6 +55,9 @@ on boot
     # Add support wakeup system from debug UART
     write /sys/devices/platform/imx-uart.0/tty/ttymxc0/power/wakeup enabled
 
+# mount the debugfs
+    mount debugfs none /sys/kernel/debug/
+
 service wpa_supplicant /system/bin/wpa_supplicant \
     -Dnl80211 -e/data/misc/wifi/entropy.bin
     class late_start
diff --git a/arm2_6sl/init.rc b/arm2_6sl/init.rc
index abe4fb4..afc8eb1 100755
--- a/arm2_6sl/init.rc
+++ b/arm2_6sl/init.rc
@@ -51,6 +51,9 @@ on boot
     # Add support wakeup system from debug UART
     write /sys/devices/platform/imx-uart.0/tty/ttymxc0/power/wakeup enabled
 
+# mount the debugfs
+    mount debugfs none /sys/kernel/debug/
+
 service wpa_supplicant /system/bin/wpa_supplicant \
     -Dnl80211 -e/data/misc/wifi/entropy.bin
     class late_start
diff --git a/imx53_smd/init.freescale.rc b/imx53_smd/init.freescale.rc
index 206f856..2fbbcc9 100755
--- a/imx53_smd/init.freescale.rc
+++ b/imx53_smd/init.freescale.rc
@@ -62,6 +62,9 @@ on boot
     chown system system /sys/class/backlight/pwm-backlight.0/brightness
     chmod 0660 /sys/class/backlight/pwm-backlight.0/brightness
 
+# mount the debugfs
+    mount debugfs none /sys/kernel/debug/
+
 # Daemon processes to be run by init.
 service hciattach /system/bin/logwrapper /system/bin/hciattach -n -s 115200 /dev/ttymxc2 ath3k 3000000 flow sleep
     user root
diff --git a/imx5x/init.rc b/imx5x/init.rc
index 048c9d0..b470988 100644
--- a/imx5x/init.rc
+++ b/imx5x/init.rc
@@ -356,9 +356,6 @@ on boot
     write /sys/class/android_usb/android0/iProduct $ro.product.model
     write /sys/class/android_usb/android0/iSerial $ro.serialno
 
-# mount the debugfs
-    mount debugfs none /sys/kernel/debug/
-
 # Prepare for RIL
     setprop gsm.ril.delay 15
     setprop ro.ril.wake_lock_timeout 300
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index ecd5017..da4ebce 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -352,9 +352,6 @@ on boot
     write /sys/class/android_usb/android0/iProduct $ro.product.model
     write /sys/class/android_usb/android0/iSerial $ro.serialno
 
-# mount the debugfs
-    mount debugfs none /sys/kernel/debug/
-
 # Set this property so surfaceflinger is not started by system_init
     setprop system_init.startsurfaceflinger 0
 
diff --git a/imx6/sabresd/init.rc b/imx6/sabresd/init.rc
index e9b5581..bb61e2c 100755
--- a/imx6/sabresd/init.rc
+++ b/imx6/sabresd/init.rc
@@ -94,6 +94,9 @@ on boot
     setprop hw.bluetooth 0
     setprop wlan.interface wlan0
 
+# mount the debugfs
+    mount debugfs none /sys/kernel/debug/
+
 service p2p_supplicant /system/bin/wpa_supplicant \
     -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf -N \
     -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf
-- 
1.8.0

